snap/info_test.go

```diff
@@ -1074,3 +1074,45 @@ func (s *infoSuite) TestNickname(c *C) {
 	c.Check(snap.DropNick("system"), Equals, "core")
 	c.Check(snap.DropNick("foo"), Equals, "foo")
 }
+
+func (s *infoSuite) TestInstanceStoreName(c *C) {
+	store, key := snap.SplitInstanceName("foo_bar")
+	c.Check(store, Equals, "foo")
+	c.Check(key, Equals, "bar")
+
+	store, key = snap.SplitInstanceName("foo")
+	c.Check(store, Equals, "foo")
+	c.Check(key, Equals, "")
+
+	store, key = snap.SplitInstanceName("_bar")
```

Comment:
this and the one after should probably come last with a note that they are not expected